### ISSUE TITLE
request help: Ingress Controller spamming info log repeatly

### ISSUE THREAD

**Issue by choryuidentify** [apache/apisix-ingress-controller#1302]
### Issue description

Hi.

I installed apisix on GKE Autopilot cluster using with this command.
```shell
helm install apisix apisix/apisix \
  --set gateway.type=LoadBalancer \
  --set ingress-controller.enabled=true \
  --set gateway.tls.enabled=true \
  --namespace ingress-apisix \
  --set ingress-controller.config.apisix.serviceNamespace=ingress-apisix
```

After deploy, `apisix-ingress-controller` start log spamming immediately.

```
2022-08-31T08:34:42+08:00       ESC[34minfoESC[0m       ingress/controller.go:606       service kube-system/vpa-recommender not found
2022-08-31T08:34:43+08:00       ESC[34minfoESC[0m       ingress/controller.go:606       service kube-system/gcp-controller-manager not found
2022-08-31T08:34:43+08:00       ESC[34minfoESC[0m       ingress/controller.go:606       service kube-system/vpa-updater not found
2022-08-31T08:34:45+08:00       ESC[34minfoESC[0m       ingress/controller.go:606       service kube-system/gcp-controller-manager not found
2022-08-31T08:34:46+08:00       ESC[34minfoESC[0m       ingress/controller.go:606       service kube-system/vpa-recommender not found
2022-08-31T08:34:47+08:00       ESC[34minfoESC[0m       ingress/controller.go:606       service kube-system/gcp-controller-manager not found
2022-08-31T08:34:47+08:00       ESC[34minfoESC[0m       ingress/controller.go:606       service kube-system/vpa-updater not found
2022-08-31T08:34:49+08:00       ESC[34minfoESC[0m       ingress/controller.go:606       service kube-system/gcp-controller-manager not found
2022-08-31T08:34:50+08:00       ESC[34minfoESC[0m       ingress/controller.go:606       service kube-system/vpa-recommender not found
2022-08-31T08:34:51+08:00       ESC[34minfoESC[0m       ingress/controller.go:606       service kube-system/gcp-controller-manager not found
2022-08-31T08:34:51+08:00       ESC[34minfoESC[0m       ingress/controller.go:606       service kube-system/vpa-updater not found
[GIN] 2022/08/31 - 08:34:52 | 200 |      55.673µs |   10.16.130.193 | GET      "/healthz"
[GIN] 2022/08/31 - 08:34:52 | 200 |      34.485µs |   10.16.130.193 | GET      "/healthz"
2022-08-31T08:34:53+08:00       ESC[34minfoESC[0m       ingress/controller.go:606       service kube-system/gcp-controller-manager not found
2022-08-31T08:34:54+08:00       ESC[34minfoESC[0m       ingress/controller.go:606       service kube-system/vpa-recommender not found
2022-08-31T08:34:55+08:00       ESC[34minfoESC[0m       ingress/controller.go:606       service kube-system/gcp-controller-manager not found
2022-08-31T08:34:55+08:00       ESC[34minfoESC[0m       ingress/controller.go:606       service kube-system/vpa-updater not found
2022-08-31T08:34:57+08:00       ESC[34minfoESC[0m       ingress/controller.go:606       service kube-system/gcp-controller-manager not found
2022-08-31T08:34:58+08:00       ESC[34minfoESC[0m       ingress/controller.go:606       service kube-system/vpa-recommender not found
2022-08-31T08:34:59+08:00       ESC[34minfoESC[0m       ingress/controller.go:606       service kube-system/gcp-controller-manager not found
2022-08-31T08:34:59+08:00       ESC[34minfoESC[0m       ingress/controller.go:606       service kube-system/vpa-updater not found
2022-08-31T08:35:01+08:00       ESC[34minfoESC[0m       ingress/controller.go:606       service kube-system/gcp-controller-manager not found
[GIN] 2022/08/31 - 08:35:02 | 200 |      61.686µs |   10.16.130.193 | GET      "/healthz"
[GIN] 2022/08/31 - 08:35:02 | 200 |      29.896µs |   10.16.130.193 | GET      "/healthz"
2022-08-31T08:35:02+08:00       ESC[34minfoESC[0m       ingress/controller.go:606       service kube-system/vpa-recommender not found
2022-08-31T08:35:03+08:00       ESC[34minfoESC[0m       ingress/controller.go:606       service kube-system/gcp-controller-manager not found
2022-08-31T08:35:03+08:00       ESC[34minfoESC[0m       ingress/controller.go:606       service kube-system/vpa-updater not found
2022-08-31T08:35:05+08:00       ESC[34minfoESC[0m       ingress/controller.go:606       service kube-system/gcp-controller-manager not found
2022-08-31T08:35:06+08:00       ESC[34minfoESC[0m       ingress/controller.go:606       service kube-system/vpa-recommender not found
2022-08-31T08:35:07+08:00       ESC[34minfoESC[0m       ingress/controller.go:606       service kube-system/gcp-controller-manager not found
2022-08-31T08:35:08+08:00       ESC[34minfoESC[0m       ingress/controller.go:606       service kube-system/vpa-updater not found
2022-08-31T08:35:09+08:00       ESC[34minfoESC[0m       ingress/controller.go:606       service kube-system/gcp-controller-manager not found
2022-08-31T08:35:10+08:00       ESC[34minfoESC[0m       ingress/controller.go:606       service kube-system/vpa-recommender not found
2022-08-31T08:35:11+08:00       ESC[34minfoESC[0m       ingress/controller.go:606       service kube-system/gcp-controller-manager not found
2022-08-31T08:35:12+08:00       ESC[34minfoESC[0m       ingress/controller.go:606       service kube-system/vpa-updater not found
[GIN] 2022/08/31 - 08:35:12 | 200 |      58.683µs |   10.16.130.193 | GET      "/healthz"
[GIN] 2022/08/31 - 08:35:12 | 200 |      32.566µs |   10.16.130.193 | GET      "/healthz"
2022-08-31T08:35:13+08:00       ESC[34minfoESC[0m       ingress/controller.go:606       service kube-system/gcp-controller-manager not found
2022-08-31T08:35:14+08:00       ESC[34minfoESC[0m       ingress/controller.go:606       service kube-system/vpa-recommender not found
2022-08-31T08:35:15+08:00       ESC[34minfoESC[0m       ingress/controller.go:606       service kube-system/gcp-controller-manager not found
2022-08-31T08:35:16+08:00       ESC[34minfoESC[0m       ingress/controller.go:606       service kube-system/vpa-updater not found
2022-08-31T08:35:17+08:00       ESC[34minfoESC[0m       ingress/controller.go:606       service kube-system/gcp-controller-manager not found
2022-08-31T08:35:18+08:00       ESC[34minfoESC[0m       ingress/controller.go:606       service kube-system/vpa-recommender not found
2022-08-31T08:35:20+08:00       ESC[34minfoESC[0m       ingress/controller.go:606       service kube-system/gcp-controller-manager not found
2022-08-31T08:35:20+08:00       ESC[34minfoESC[0m       ingress/controller.go:606       service kube-system/vpa-updater not found
2022-08-31T08:35:22+08:00       ESC[34minfoESC[0m       ingress/controller.go:606       service kube-system/gcp-controller-manager not found
[GIN] 2022/08/31 - 08:35:22 | 200 |       70.52µs |   10.16.130.193 | GET      "/healthz"
[GIN] 2022/08/31 - 08:35:22 | 200 |       82.59µs |   10.16.130.193 | GET      "/healthz"
2022-08-31T08:35:22+08:00       ESC[34minfoESC[0m       ingress/controller.go:606       service kube-system/vpa-recommender not found
2022-08-31T08:35:24+08:00       ESC[34minfoESC[0m       ingress/controller.go:606       service kube-system/vpa-updater not found
2022-08-31T08:35:24+08:00       ESC[34minfoESC[0m       ingress/controller.go:606       service kube-system/gcp-controller-manager not found
2022-08-31T08:35:26+08:00       ESC[34minfoESC[0m       ingress/controller.go:606       service kube-system/gcp-controller-manager not found
2022-08-31T08:35:26+08:00       ESC[34minfoESC[0m       ingress/controller.go:606       service kube-system/vpa-recommender not found
2022-08-31T08:35:28+08:00       ESC[34minfoESC[0m       ingress/controller.go:606       service kube-system/vpa-updater not found
2022-08-31T08:35:28+08:00       ESC[34minfoESC[0m       ingress/controller.go:606       service kube-system/gcp-controller-manager not found
2022-08-31T08:35:30+08:00       ESC[34minfoESC[0m       ingress/controller.go:606       service kube-system/gcp-controller-manager not found
2022-08-31T08:35:30+08:00       ESC[34minfoESC[0m       ingress/controller.go:606       service kube-system/vpa-recommender not found
[GIN] 2022/08/31 - 08:35:32 | 200 |      47.964µs |   10.16.130.193 | GET      "/healthz"
[GIN] 2022/08/31 - 08:35:32 | 200 |      52.842µs |   10.16.130.193 | GET      "/healthz"
2022-08-31T08:35:32+08:00       ESC[34minfoESC[0m       ingress/controller.go:606       service kube-system/vpa-updater not found
2022-08-31T08:35:32+08:00       ESC[34minfoESC[0m       ingress/controller.go:606       service kube-system/gcp-controller-manager not found
2022-08-31T08:35:34+08:00       ESC[34minfoESC[0m       ingress/controller.go:606       service kube-system/gcp-controller-manager not found
2022-08-31T08:35:34+08:00       ESC[34minfoESC[0m       ingress/controller.go:606       service kube-system/vpa-recommender not found
2022-08-31T08:35:36+08:00       ESC[34minfoESC[0m       ingress/controller.go:606       service kube-system/vpa-updater not found
2022-08-31T08:35:36+08:00       ESC[34minfoESC[0m       ingress/controller.go:606       service kube-system/gcp-controller-manager not found
2022-08-31T08:35:38+08:00       ESC[34minfoESC[0m       ingress/controller.go:606       service kube-system/gcp-controller-manager not found
2022-08-31T08:35:38+08:00       ESC[34minfoESC[0m       ingress/controller.go:606       service kube-system/vpa-recommender not found
2022-08-31T08:35:40+08:00       ESC[34minfoESC[0m       ingress/controller.go:606       service kube-system/vpa-updater not found
2022-08-31T08:35:40+08:00       ESC[34minfoESC[0m       ingress/controller.go:606       service kube-system/gcp-controller-manager not found
```

It's a bug? or can I disable it?

Thank you.

### Environment

- your apisix-ingress-controller version (output of apisix-ingress-controller version --long):
Version: 1.4.1
Git SHA: no-git-module
Go Version: go1.16.15
Building OS/Arch: linux/amd64
Running OS/Arch: linux/amd64

- your Kubernetes cluster version (output of kubectl version):
Client Version: version.Info{Major:"1", Minor:"24", GitVersion:"v1.24.3", GitCommit:"aef86a93758dc3cb2c658dd9657ab4ad4afc21cb", GitTreeState:"clean", BuildDate:"2022-07-13T14:30:46Z", GoVersion:"go1.18.3", Compiler:"gc", Platform:"linux/amd64"}
Kustomize Version: v4.5.4
Server Version: version.Info{Major:"1", Minor:"22", GitVersion:"v1.22.10-gke.600", GitCommit:"2c921d7b040ed9c5a3a1f9407fb109b74d72d0a4", GitTreeState:"clean", BuildDate:"2022-06-02T09:20:24Z", GoVersion:"go1.16.15b7", Compiler:"gc", Platform:"linux/amd64"}

- if you run apisix-ingress-controller in Bare-metal environment, also show your OS version (uname -a):
GKE Autopilot

**Comment by tao12345666333** [comment]
APISIX Ingress will watch all namespaces by default. You can change the namespace selector configuration to filter namespace

**Comment by choryuidentify** [comment]
@tao12345666333 Thanks!

**Comment by amartincolby** [comment]
Hi. I encountered this same issue and solved it as described. The second issue I found is that if no namespace labels match the `namespaceSelector`, then it defaults to using `appNamespaces`. This was confusing to me since I would have assumed that if no namespace labels match the `namespaceSelector`, then no namespaces should be watched.

**Comment by tao12345666333** [comment]
> Hi. I encountered this same issue and solved it as described. The second issue I found is that if no namespace labels match the `namespaceSelector`, then it defaults to using `appNamespaces`. This was confusing to me since I would have assumed that if no namespace labels match the `namespaceSelector`, then no namespaces should be watched.

This logic is fixed in our upcoming v1.5.0 release. The release has entered the release window, and voting is currently underway. will be published soon https://lists.apache.org/thread/tmmlbys8y82t9v7kf4xjzhjxvp2lsqlt

**Comment by wofr** [comment]
As linked above I observed the same behaviour.  I'm quite uncertain how changing the namespace helps me, as the problem is that the specific services are not found while scanning all-namespaces. 

I think I missed the point ;(
